### ISSUE TITLE
Uzupełnianie numerów bocznych/typów pojazdów

### ISSUE THREAD

**Issue by p0358** [jacekkow/mpk-ttss#19]
Z tego co widzę pod https://mpk.jacekk.net/vehicles/ zaczęły się pojawiać pierwsze pojazdy, w związku z tym otwieram ten issue, by pomóc w uzupełnianiu listy wspólnymi siłami. Tyle udało mi się zabrać dotychczas:

HL425 t-1188950297274252828
HZ264 t-1188950297274256062
HW147 t-1188950297274253264
??534 b-1152921495817824215
PR793 b-1152921495817823054
Krakowiak 920 t-1188950297274255996
HL439 t-1188950297274252775
HG929 t-1188950297274252286
RZ212 t-1188950297274253140
RF303 t-1188950297274253000
HK454 t-1188950297274252719
HL434 t-1188950297274252796
HG909 t-1188950297274252355
RG913 t-1188950297274252339

**Comment by p0358** [comment]
HL413 t-1188950297274252861
HG926 t-1188950297274252294

![](https://i.imgur.com/pXkp0zK.jpg)
^ choć muszę przyznać, że to nie pomaga

**Comment by domints** [comment]
Jak można wprowadzać dane do tej listy? Widziałem, że kiedyś był formularz, czy teraz też jakiś jest? Mógłbym wieczorami po powrocie do domu batchem wrzucić wszystko co udało mi się zebrać moją aplikacją :)

**Comment by domints** [comment]
Rozumiem, Jacku, że nie będziesz miał problemu, jak wieczorami wejdzie automat, który może wysłać (jak będę miał czas pospisywać) może nawet ponad 100 pozycji jedna za drugą? Czy może wolałbyś, żebym robił to z jakimś odstępem, np. co 100ms żeby niepotrzebnie nie obciążać Ci serwera?

**Comment by jacekkow** [comment]
@domints: Jeśli tylko będzie sekwencyjnie, a nie równolegle - nie ma problemu.

**Comment by p0358** [comment]
To będę od teraz wysłał przez formularz, jedynie jeszcze bym zgłosił, że HZ271 jest wpisany na liście jako HZ270, trzeba poprawić

**Comment by jacekkow** [comment]
@p0358: Jak dla mnie jest w porządku - to jest zespół dwóch wagonów i pierwszy ma numer HZ270:
https://matel4472.flog.pl/wpis/12547375

**Comment by domints** [comment]
@p0358 tak, dokładnie jak Jacek mówi, do grudnia te ID zawsze wskazywały na numer taborowy pierwszego członu (czy też wagonu silnikowego w przypadku wiedni) i myślę, że do tej pory tak jest.

**Comment by p0358** [comment]
Zastanawiałem się nad tym w jaki sposób można by było programowo wyliczać numery boczne pojazdów, poczyniłem kilka notatek:

> HL437 t-11889502972 74252784 (31 stycznia)
> HL437 t-11889502972 62353836 (12 lutego)
> HL437 t-11889502972 58552006 (13 lutego)
> 31-12 (13 dni)  -> 11898948 [915303,6923076923/dzień]
> 12-13 (1 dzień) -> 3801830 [?]
> 
> HL413 t-11889502972 74252861 (31 stycznia)
> HL413 t-11889502972 69201267 (4 lutego)
> 31-4 (5 dni)    -> 5051594 [1010318,8/dzień]
> 
> Jednego dnia:
> HL413 t-11889502972 74252861 HL437 t-11889502972 74252784
> diff=77
>
> Powtarzalne:
> 
> HW147 t-1188950297274253264 (31 stycznia)
> HW147?+HB547 t-1188950297274253264 (30 stycznia)
> 

Niestety mam za mało powtarzających się pojazdów, by wyciągnąć konkretne wnioski, mogę na pewno powiedzieć, że ID z dnia na dzień się konsekwentnie zmniejszają.
@jacekkow Mógłbyś udostępnić wszystkie swoje zebrane mapowania tramwajów? Jakby były z dokładnymi datami dodania, byłoby idealnie.
Jeżeli okazałoby się, że offsety/różnice między poszczególnymi pojazdami pozostają identyczne, wystarczyłoby zapisać aktualne ID dowolnego pojazdu, by zaktualizować mapowania dla wszystkich, później pozostałoby tylko zastanowić się na jakiej podstawie one się zmniejszają/napisać jakiś kod, który automatycznie na jakiejś podstawie by te zmiany wykrywał. (według ostatniej części notatki, zmieniają się one falami, a tak pozostają stałe przez jakiś czas)

**Comment by jacekkow** [comment]
@p0358 Dane tutaj - podzieliłem ręcznie na kolejne "generacje" (być może część podziałów jest sztuczna i pliki powinny być połączone), ale dokładnych dat nie mam: https://gist.github.com/jacekkow/78d1fac3b31be9713ef264d9569758ec

Nie mam pewności, że wszystkie dane są poprawne - mogą być drobne błędy.

> Jeżeli okazałoby się, że offsety/różnice między poszczególnymi pojazdami pozostają identyczne

Oszczędzę szukania - nie są.

```
./ttss-2018-4.txt:-1188950297309646177|RY820
./ttss-2019-1.txt:-1188950297274252427|RY820
./ttss-2019-4.txt:-1188950297269198342|RY820

offset 2018-4 2019-1 35393750
offset 2019-1 2019-4 5054085
offset 2018-4 2019-4 40447835


./ttss-2018-4.txt:-1188950297309646346|RP615
./ttss-2019-4.txt:-1188950297269200383|RP615

offset 2018-4 2019-4 40445963


./ttss-2018-4.txt:-1188950297309646273|RP635
./ttss-2019-1.txt:-1188950297274252535|RP635
./ttss-2019-4.txt:-1188950297269201120|RP635

offset 2018-4 2019-1 35393738
offset 2019-1 2019-4 5051415
offset 2018-4 2019-4 40445153
```

Niestety czasami różnice między kolejnymi tramwajami wynoszą cztery, a czasami pięć. Wystarczy parę takich zmian kilka i nagle przesuwamy się nie tylko o parę numerów, a o całe serie. Część danych jest ładnie poukładana - druga część wygląda jak "sieczka" (stąd wydaje mi się, że ttss-2019 cz. 1 i 2 pochodzą z jednego czasookresu).


> kod, który automatycznie na jakiejś podstawie by te zmiany wykrywał

To akurat proste - numery zmieniają się o ładne parę tysięcy do przodu, a stare nie występują.
